kernel.go

```diff
@@ -43,18 +44,38 @@ type SocketGroup struct {
 	Key           []byte
 }
 
-// kernelInfo holds information about the igo kernel, for
-// kernel_info_reply messages.
-type kernelInfo struct {
-	ProtocolVersion []int  `json:"protocol_version"`
-	Language        string `json:"language"`
-}
-
 // kernelStatus holds a kernel state, for status broadcast messages.
 type kernelStatus struct {
 	ExecutionState string `json:"execution_state"`
 }
 
+// KernelLanguageInfo holds information about the language that this kernel executes code in
```

Comment:
Comment should be a complete sentence (i.e., including periods etc.).  This is because these comments turn into the GoDocs.  Minor thing, but should be consistent throughout.